docs/DeveloperGuide.adoc

```diff
@@ -262,6 +330,239 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
 
 Certain properties of the application can be controlled (e.g user prefs file location, logging level) through the configuration file (default: `config.json`).
 
+=== Module Tagging Feature
+
+==== Implementation
+The module tagging mechanism is facilitated by `UserTag`, `DefaultTag` and `UniqueTagList`.
+
+`UserTag` and `DefaultTag` implement the `Tag` interface. The former represents user-created tags while the latter
+represents default tags of one of the types in `DefaultTagType`. The key difference between `UserTag` and `DefaultTag`
+is that the first is user-modifiable while the second is not and is essentially immutable. The operation
+`Tag#isDefault()` is implemented to differentiate between the two classes of `Tag`.
+
+Each `Tag` is stored in a `UniqueTagList`, which implements `Iterable<Tag>` and is stored internally in `Module` and
+`StudyPlan` as `tags`.  It represents a list of tags that can be of type `UserTag` or `DefaultTag` and enforces the
+uniqueness between the tags. Additionally, it implements the following operations:
+
+- `UniqueTagList#addTag(Tag toAdd)` -- Adds the given `Tag` to the list.
+- `UniqueTagList#removeTag(Tag toRemove)` -- Removes the given `Tag` from the list.
+- `UniqueTagList#containsTagWithName(String tagName)` -- Checks if the list contains a `Tag` with the given `tagName`.
+- `UniqueTagList#getTag(String tagName)` -- Returns the `Tag` with the given `tagName`.
+
+Given below is an example usage scenario of the `UniqueTagList#addTag(Tag toAdd)` operation and how the module tagging
+mechanism behaves at each step.
+
+Step 1. The user executes the `addtag MODULE_CODE TAG_NAME`. command to add a tag with the name _TAG_NAME_ to the
+module with the module code _MODULE_CODE_.  A `toAdd` variable of type `Tag` and a boolean value `newTagCreated` is
+maintained during the execution of the command to represent the tag that is to be added and whether or not a new tag
+has been created respectively. The `addtag` command calls `Model#activeSpContainsTag(String tagName)` to check if the
+active study plan contains a tag with the given `tagName`. This method accesses the active study plan in the module
+planner and checks if such a tag exists in its `UniqueTagList`. There are two possible scenarios that are described
+in steps 2 and 3.
+
+Step 2. If such a tag does not exist, a new `UserTag` is created and is assigned to `toAdd`.
+
+image::TagModuleCommandDiagram1.png[]
```

Comment:
Would it be better if you keep the style of the diagram (i.e. colour, border style, font etc.) consistent with the rest of the document?